### PR TITLE
GROUP MANAGEMENT: Adds logic for view-mode and edit-mode for groups data

### PR DESCRIPTION
**Pull request by njons**
#### Description:
Adds UI functionality: Switching from viewing data in table to entering edit mode where all interactions related to the data are handled. This PR only adds logic for the render switch.   

>**All elements and text are hard coded for now and styling is inline for mocking purposes. No functionality beyond that already existing (and the one described above) have been added.**

```
Current structure of group management: 
 <Groups />
   <WizardPanel/>                     handles logic: what wizard to show at what time (currently only showing help to make first group)
       - RenderFirstGroupWizard()     placeholder function (could become component)
   <DataPanel/>                       handles logic: viewing data vs editing data 
       - ViewDataTable()              placeholder function (could become component)
       - EditDataBox()                placeholder function (could become component)
```

**Summary**

1. Edits to previous preparations:
   - placeholder `RenderDataPanel()` function in `<Groups/>` replaced with `<DataPanel/>` 
      - Added in: `login/components/DataPanel/`
   - `<DataPanel>` has got its own folder as it represents a new concept of dividing viewing data and only interacting with data in a dedicated panel and thus has been kept separate form current files

2. Adding logic (essentially the same as for `<WizardPanel/>`, PR #420): 
-  before state change in`<DataPanel/>`:
    - `state.editDataMode: false` > renders `RenderViewData()` >  `<ViewDataTable/>` functional  child component > lists group names, each with an `edit` button
- trigger in child (`<ViewDataTable/>`) to change state in parent (`<DataPanel/>`): 
   -  clicking `edit` button  > triggers change in `<DataPanel/>`: `state.editDataMode: true`
- after state change in parent  in`<DataPanel/>`: 
   - `state.editDataMode: true` > renders `RenderEditData ` >  `<EditDataBox />`  functional  child component > offers navigation through all editing options for the available group data

:warning:   **Previous PR introduced a wizard which will be rendered on load. However, in final functionality the wizard to help users create their first group will not be visible if the user has created their own group.**

---
###### Groups data view-mode (state.editDataMode: false)
Screenshot 2020-09-03 at 12 10 00

###### Groups data edit-mode  (state.editDataMode: true)
Screenshot 2020-09-03 at 12 08 29

---

#### For reviewer:
**WHAT?**
- [x] Check that clicking `edit` on the group table renders the edit mode panel

**HOW?**
- [ ] Read the above description
- [ ] Reviewed the code changes
- [ ] Navigate to the branch (pulled the latest version)
- [ ] Run the code and been able to execute the expected function
- [ ] Check any styling on both desktop and mobile sizes

### CODE REVIEW

File: src/login/components/DataPanel/DataPanel.js [hunk 1]
```diff
@@ -67,7 +85,10 @@ class DataPanel extends Component {
           margin: "1rem 0 ",
         }}
       >
-        <RenderViewData editDataMode={this.state.editDataMode} />
+        <RenderViewData
```
Comment (eunjuhuss) [inline src/login/components/DataPanel/DataPanel.js]: would it be possible to combine the RenderViewdata/RenderEditData into one?

```
<RenderData
        toggleEditMode={this.toggleEditMode}
         editDataMode={this.state.editDataMode}
 /> 
```

```
const RenderData = (props) => {
  return (
    <Fragment>{props.editDataMode ? <EditDataBox {...props} /> : <ViewDataTable {...props} />}</Fragment>
  );
};

```
it's only my suggestion to reduce code, you might be have some reason for separating functions.

**Review by eunjuhuss — COMMENTED** [review]
looks good! 
I have a suggestion to combine the RenderViewdata/RenderEditData into one.

File: src/login/components/DataPanel/DataPanel.js (hunk 1 above)
Comment (njons) [reply src/login/components/DataPanel/DataPanel.js]: yes, that's a great idea! 
But you are right, I am keeping the functions separate while the other elements and components are still in development. However, if these don't need more work, combining them is an excellent suggestion.

Comment (eunjuhuss) [reply src/login/components/DataPanel/DataPanel.js]: perfect!

**Pull request merged**